test/disable-user-indexes/user-indexes-disabled.td

```diff
@@ -125,3 +133,104 @@ on_name   key_name        seq_in_index  column_name  expression  nullable enable
 t         t_secondary_idx 1             <null>       "a + a"     true     false
+
+# 🔬 Enabling indexes
+
+# 🔬🔬 Views + Sources
+
+> ALTER INDEX logging_derived_mat_primary_idx SET ENABLED;
+
+> SELECT * FROM logging_derived_mat
+1
+
+> SHOW INDEXES FROM logging_derived_mat;
+on_name             key_name                        seq_in_index  column_name expression  nullable enabled
+----------------------------------------------------------------------------------------------------------
+logging_derived_mat logging_derived_mat_primary_idx 1             count       <null>      false    true
+
+# Alter index is functionally idempotent
+> ALTER INDEX logging_derived_mat_primary_idx SET ENABLED;
+
+> SELECT * FROM logging_derived_mat
+1
+
+# 🔬🔬🔬 Materialized views
+
+> ALTER INDEX mat_view_primary_idx  SET ENABLED;
+
+> SHOW INDEXES FROM mat_view;
+on_name   key_name              seq_in_index  column_name  expression  nullable enabled
+---------------------------------------------------------------------------------------
+mat_view  mat_view_primary_idx  1             sum          <null>      true     true
+
+> SHOW MATERIALIZED VIEWS
+logging_derived_mat
+mat_view
+
+> SELECT * FROM mat_view
+1
+
+# 🔬🔬🔬 Materialized sources
+
+> ALTER INDEX mat_data_primary_idx  SET ENABLED;
+
+> SHOW INDEXES FROM mat_data;
+on_name   key_name              seq_in_index  column_name  expression  nullable enabled
+---------------------------------------------------------------------------------------
+mat_data  mat_data_primary_idx  1             a            <null>      false    true
+
+> SHOW MATERIALIZED SOURCES
+mat_data
+
+> SELECT * FROM mat_data
+1
+
+# 🔬🔬🔬 Non-materialized views
+
+# Still cannot select from non-materialized view because some of its indexes are disabled
+
+! SELECT * FROM join_view
+Unable to automatically determine a timestamp for your query; this can happen if your query depends on non-materialized sources
+
+> ALTER INDEX join_data_primary_idx SET ENABLED
+
+> SELECT * FROM join_view;
+1 a
+
+# 🔬🔬 Sinks
+
+# With indexes enabled, data starts flowing to sinks
+$ kafka-verify format=avro sink=materialize.public.snk_indexes_disabled sort-messages=true
+{"before": null, "after": {"row":{"a": 1}}}
+
+# 🔬🔬 Tables
+
+# Tables still work with non-primary indexes enabled because all indexes are covering
+> ALTER INDEX t_secondary_idx SET ENABLED;
+
+> SHOW INDEXES FROM t;
+on_name   key_name        seq_in_index  column_name  expression  nullable enabled
+-----------------------------------------------------------------------------
+t         t_primary_idx   1             a            <null>      true     false
+t         t_secondary_idx 1             <null>       "a + a"     true     true
+
+> INSERT INTO t VALUES (1)
+
+> SELECT * FROM t
+1
+
+# Check idempotency
+> ALTER INDEX t_secondary_idx SET ENABLED;
+
+> INSERT INTO t VALUES (2)
+
+> SELECT * FROM t
+1
+2
+
+# First index re-enabled becomes table's primary index and can no longer be dropped
```

Comment:
Is this what we want? Are there things that assume the default index is the initial primary idx, or that it always exists?

Reply:
Ah, the table's primary index will—I think, check me on this—always be the index on that table with the lowest ID. Right? It's always created immediately after the table, in the same transaction—in fact the ID should always be _exactly_ one greater than the table's ID, though I wouldn't rely on that.

If that holds, I think it would be great if we could prevent you from enabling a table's secondary index until its primary index is enabled. The idea being that once tables are persistence, enabling the primary index will cause the table to actually be loaded from disk, which you've got to do before you can build any non-primary indexes.

Reply:
Ah, compelling. Will do that.